LocalStoreSimulation/src/entities/Product.java

```diff
@@ -0,0 +1,80 @@
+package entities;
+
+import java.util.Objects;
+
+public class Product {
+    private static int nextId = 1;
+    private int id;
+    private String type;
+    private String name;
+    private int price;
+    private int weight;
+    private int quantity;
+    private String validityDate;
+
+    public Product() {
+    }
+
+    public Product(int id, int quantity) {
+        this.id = id;
+        this.quantity = quantity;
+    }
+
+    public Product(int id, String name, int price, int weight, int quantity, String validityDate) {
+        this.id = Product.nextId;
+        Product.nextId++;
```

Comment:
++ operator increments the value after it's used. So you can merge these 2 lines into a single one:
this.id = Product.nextId++;